src/vulkan.cpp

```diff
@@ -724,17 +745,78 @@ void Vulkan::createCommandBuffer() {
 
         commandBuffer.end();
     }
+    commandBuffersForNoPresent.resize(2);
```

Comment:
I would probably move the command buffer recording for the  `commandBuffersForNoPresent` in a separete function.

Also, the code duplication feels pretty ugly - is it possible to wrap those logical blocks (which are currently separated by comments such as `// RAY TRACING`) into dedicated functions, which can then be called for the respective command buffers (`commandBuffers` and `commandBuffersForNoPresent`)?

Reply:
When I try to move these command buffer, I find that there is a memory barrier that move image layout from undefined to general.
```// RENDER TARGET IMAGE & SUMMED PIXEL COLOR IMAGE: UNDEFINED -> GENERAL```
From my opinion, summed pixel color image should not change layout because it only be read and write by compute shader. And from undefined layout to other layout would not preserve image's previous data.
Summed pixel color image should be initialized to zero and change layout to general, then it should only use memory barrier to sync two compute shaders' write and read.